### PR TITLE
tech detection & retire: Remove usage of CWE-200 and update other CWE

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOGs > Add note.
- Rules > Changed or dropped CWE.
- Unit Tests > Updated for the new or removed CWEs.

## Related Issues
- Fixes zaproxy/zaproxy#8732
- Fixes zaproxy/zaproxy#8733

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

**Comment by psiinon** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [563c3df4\-e28b\-4589\-8ec3\-dbfbd11b082f](https://eu.ast.checkmarx.net/projects/0da811fa-6373-42b1-bbb6-d0026285b136/scans?id=563c3df4-e28b-4589-8ec3-dbfbd11b082f&branch=kingthorin%2Fwapp-ret-cwe)
### No New Or Fixed Issues Found

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPassiveScanner.java [hunk 1]
```diff
@@ -433,7 +433,9 @@ Builder createAlert(String url, ApplicationMatch appMatch) {
                 .setConfidence(Alert.CONFIDENCE_MEDIUM)
                 .setUri(url)
                 .setDescription(getDesc(app))
-                .setCweId(200)
+                .setCweId(
+                        497) // CWE-497: Exposure of Sensitive System Information to an Unauthorized
```
Comment (thc202) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPassiveScanner.java:437]: Not sure we are checking for sensitive system nformation but if there's no better ID.

**Comment by thc202** [comment]
This has conflicts.

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPassiveScanner.java (hunk 1 above)
Comment (thc202) [reply addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPassiveScanner.java:437]: Maybe that's better.

**Comment by kingthorin** [comment]
Tweaked

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by thc202** [comment]
Commit message needs tweak though (it has conflict report and to match the change in the tech detect).

**Comment by kingthorin** [comment]
Done

**Review by ricekot — APPROVED** [review]

**Pull request merged**